src/components/error-boundary/error-boundary.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { ReactNode, ErrorInfo } from 'react';
+import { formatMessage } from '../../utility/errors';
+import { AlertDialog, AlertDialogProps } from '../alert-dialog/alert-dialog';
+
+
+export interface AlertErrorProps {
+  message?: string,
+  alertTitle?: string,
+  error?: Error,
+  info?: ErrorInfo,
+}
+
+export class AlertError extends Error {
+  readonly alertTitle?: string;
+  readonly error?: Error;
+  readonly info?: ErrorInfo;
+
+  constructor(props: AlertErrorProps) {
+    super(props.message);
+    this.alertTitle = props.alertTitle;
+    this.error = props.error;
+    this.info = props.info;
+  }
+}
+
+
+export interface ErrorBoundaryProps {
+  children?: ReactNode,
+  alertTitle?: string,
+  errorMessage?: string,
+  alertDialogProps: AlertDialogProps,
+  setAlertDialogProps: (props: AlertDialogProps) => void,
+}
+
+export interface ErrorBoundaryState {
+  hasError: boolean,
```

Comment:
Can `hasError` be removed? It looks like it's redundant since `error` is optional and can be checked to determine the same thing.

Reply:
Definitely!

Reply:
Resolved in the latest commit.